--- process.c	2003-09-18 21:53:54.000000000 +0900
+++ /home/lowid/programming/exam-source/screen-4.0.2/process.c	2007-11-16 14:18:14.000000000 +0900
@@ -1687,11 +1687,42 @@
 	}
       /* FALLTHROUGH */
     case RC_OTHER:
-      if (MoreWindows())
-	SwitchWindow(display && D_other ? D_other->w_number : NextWindow());
-      break;
+    if (MoreWindows()){
+        /* default(no arguments) */
+        if ((atoi(args[0]) == 0) && (atoi(args[1]) == 0))
+            SwitchWindow(display && D_fore? D_other->w_number : NextWindow());
+        else if((atoi(args[0]) == 0) && (atoi(args[1]) != 0)){
+            if(atoi(args[1]) != D_fore->w_number)
+                SwitchWindow(atoi(args[1]));
+            else if(atoi(args[1]) == D_fore->w_number){
+                /* 검사할때 같아서 그럼 SwitchWindow 쓰지말고 이케 해야함 */
+                SetForeWindow(wtab[D_other->w_number]);
+                Activate(fore->w_norefresh);  
+            }
+        }
+        /* 현재 윈도우가 첫번째,두번째 인자의 윈도우에 둘다 위치하지 않는다면 
+         * 첫번째 인자의 윈도우 넘버로 이동 */
+        else if((atoi(args[0]) != D_fore->w_number) && (atoi(args[1]) != D_fore->w_number))
+            SwitchWindow(atoi(args[0]));
+        /* 현재 윈도우가 첫번째 인자가 가지고 있는 숫자의 윈도우에 있다면
+         * 첫번째 인자의 윈도우 넘버로 이동 */
+        else if(atoi(args[0]) == D_fore->w_number)
+            SwitchWindow(atoi(args[1]));
+        /* 현재 윈도우가 두번째 인자가 가지고 있는 숫자의 윈도우에 있다면 
+         * 두번째 인자의 윈도우 넘버로 이동 */
+        else if(atoi(args[1]) == D_fore->w_number)
+            SwitchWindow(atoi(args[0]));
+    }
+  /* Note 
+     * 1,arg[0]는 other 다음 숫자부터
+     * 2,D_fore은 현재 위치하는 윈도우구조체를 나타내고
+     * w_number는 현재 윈도우숫자 w_title는 제목을 나타낸다 
+     * 3,Switch윈도우는 해당 윈도우로 이동시키는 역활을 하는 함수이다 */
+    /* original code 
+     * SwitchWindow(display && D_other ? D_other->w_number : NextWindow()); */
+    break;
     case RC_META:
-      if (user->u_Esc == -1)
+    if (user->u_Esc == -1)
         break;
       ch = user->u_Esc;
       s = &ch;
@@ -3156,6 +3187,7 @@
 	        i -=  T_CAPS;
 	      newact = df ? &dmtab[i] : mf ? &mmtab[i] : &umtab[i];
 	    }
+       /* Get new nr value*/
 	  if (args[1])
 	    {
 	      if ((newnr = FindCommnr(args[1])) == RC_ILLEGAL)
@@ -3163,9 +3195,13 @@
 		  Msg(0, "%s: bindkey: unknown command '%s'", rc_name, args[1]);
 		  break;
 		}
+          /* check argument (referance : comm.h) */
 	      if (CheckArgNum(newnr, args + 2) < 0)
-		break;
+		    break;
+          /* check arg 
+          Msg(0," %s %s move1:%s move2:%s %d",args[0],args[1],args[2],args[3],argc); */
 	      ClearAction(newact);
+          /* newnr ->(save) newact*/
 	      SaveAction(newact, newnr, args + 2, argl + 2);
 	      if (kf == 0 && args[1])
 		{
@@ -4631,6 +4667,7 @@
   return pp - wtab;
 }
 
+/* This!! */
 static int
 MoreWindows()
 {
